src/socket.c

```diff
@@ -284,8 +284,15 @@ gint socket_init(gint argc, gchar **argv)
 	GdkDisplay *display = gdk_display_get_default();
 	gchar *p;
 
+	/* On OS X with quartz backend gdk_display_get_name() returns hostname
+	 * using [NSHost currentHost] (it could return more or less whatever string
+	 * as display name is a X11 specific thing). This call can lead to network
+	 * query and block for several seconds so better skip it. */
+#ifndef GDK_WINDOWING_QUARTZ
```

Comment:
Well, I didn't do this in Scintilla because of problems

https://sourceforge.net/p/scintilla/bugs/1696/

neither in Geany

https://github.com/geany/geany/blob/master/src/editor.c#L4927

before already and I'm not sure if it makes much sense for OS X.